vscode/src/workspace.ts

```diff
@@ -342,67 +363,49 @@ export class Workspace implements WorkspaceInterface {
     return result;
   }
 
-  private registerRestarts(context: vscode.ExtensionContext) {
-    this.createRestartWatcher(context, "Gemfile.lock");
-    this.createRestartWatcher(context, "gems.locked");
-    this.createRestartWatcher(context, "**/.rubocop.yml");
-    this.createRestartWatcher(context, ".rubocop");
+  private registerRestarts() {
+    this.createRestartWatcher(`{${WATCHED_FILES.join(",")}}`);
 
     // If a configuration that affects the Ruby LSP has changed, update the client options using the latest
     // configuration and restart the server
-    context.subscriptions.push(
+    this.context.subscriptions.push(
       vscode.workspace.onDidChangeConfiguration(async (event) => {
         if (event.affectsConfiguration("rubyLsp")) {
-          // Re-activate Ruby if the version manager changed
-          if (
-            event.affectsConfiguration("rubyLsp.rubyVersionManager") ||
-            event.affectsConfiguration("rubyLsp.bundleGemfile") ||
-            event.affectsConfiguration("rubyLsp.customRubyCommand")
-          ) {
-            await this.ruby.activateRuby();
-          }
-
-          this.outputChannel.info(
-            "Restarting the Ruby LSP because configuration changed",
-          );
-          await this.restart();
+          await this.debouncedRestart("configuration changed");
```

Comment:
We should stop invoking `restart` directly. The debounced version guarantees that we won't perform a double restart if two independent reasons trigger them.